ccl/multiregionccl: TestMrSystemDatabase failed

ccl/multiregionccl.TestMrSystemDatabase [failed](https://tanzanite.cluster.engflow.com/invocations/default/9da80718-4636-4bcb-bc9a-b35e2374de6f?testReportRun=10&testReportShard=3&testReportAttempt=1#targets-Ly9wa2cvY2NsL211bHRpcmVnaW9uY2NsOm11bHRpcmVnaW9uY2NsX3Rlc3Q=) on master @ [4f31fbae529e459c00ef2490e193c0c246e8d0f9](https://github.com/cockroachdb/cockroach/commits/4f31fbae529e459c00ef2490e193c0c246e8d0f9):


```
=== RUN   TestMrSystemDatabase
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestMrSystemDatabase1666374437
    test_log_scope.go:81: use -show-logs to present logs inline

pkg/ccl/multiregionccl/multiregion_system_table_test.go:60: (TestMrSystemDatabase)
	NOTICE: .Locality() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().Locality() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
    multiregion_system_table_test.go:255: -- test log scope end --
test logs left over in: outputs.zip/logTestMrSystemDatabase1666374437
--- FAIL: TestMrSystemDatabase (6.19s)
=== RUN   TestMrSystemDatabase/QueryByEnum
    multiregion_system_table_test.go:252: error scanning '&{0xc009db46c0 <nil>}': pq: unsupported comparison: bytes to crdb_internal_region
    --- FAIL: TestMrSystemDatabase/QueryByEnum (0.00s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=10</code>
 - <code>shard=3</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMrSystemDatabase.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34979

--- a/pkg/ccl/multiregionccl/multiregion_system_table_test.go
+++ b/pkg/ccl/multiregionccl/multiregion_system_table_test.go
@@ -64,13 +64,8 @@ func TestMrSystemDatabase(t *testing.T) {
 	tDB := sqlutils.MakeSQLRunner(tenantSQL)
 
 	// Generate stats for system.sqlinstances. See the "QueryByEnum" test for
-	// details. Since stats are generated asynchronously, we also poll to make
-	// sure the stats are generated.
+	// details.
 	tDB.Exec(t, `ANALYZE system.sqlliveness;`)
-	tDB.CheckQueryResultsRetry(t,
-		`select count(*) from [show statistics for table system.sqlliveness] where column_names = '{crdb_region}';`,
-		[][]string{{"1"}},
-	)
 
 	tDB.Exec(t, `ALTER DATABASE system SET PRIMARY REGION "us-east1"`)
 	tDB.Exec(t, `ALTER DATABASE system ADD REGION "us-east2"`)
@@ -239,18 +234,27 @@ func TestMrSystemDatabase(t *testing.T) {
 	t.Run("QueryByEnum", func(t *testing.T) {
 		// This is a regression test for a bug triggered by setting up the system
 		// database. If the operation to configure the does not clear table
-		// statistics, this query will fail in the optimizer, because the stats
-		// will have the wrong type for the crdb_column.
-		row := tDB.QueryRow(t, `
-			SELECT crdb_region, session_id, expiration 
-			FROM system.sqlliveness 
-			WHERE crdb_region = 'us-east1'
-			LIMIT 1;`)
-		var sessionID string
-		var crdbRegion string
-		var rawExpiration apd.Decimal
-		row.Scan(&crdbRegion, &sessionID, &rawExpiration)
-		require.Equal(t, "us-east1", crdbRegion)
+		// statistics, this query will fail in the optimizer, because the stats will
+		// have the wrong type for the crdb_region column. Since stats are generated
+		// asynchronously, we poll for the results until they are correct.
+		testutils.SucceedsSoon(t, func() error {
+			var sessionID string
+			var crdbRegion string
+			var rawExpiration apd.Decimal
+			err := tenantSQL.QueryRow(`
+				SELECT crdb_region, session_id, expiration 
+				FROM system.sqlliveness 
+				WHERE crdb_region = 'us-east1'
+				LIMIT 1;`).Scan(&crdbRegion, &sessionID, &rawExpiration)
+			if err != nil {
+				return err
+			}
+			if crdbRegion != "us-east1" {
+				return errors.Newf("expected region to be us-east1; got %s", crdbRegion)
+			}
+			return nil
+		})
+
 	})
 }
 

